ndn_app/node.py

```diff
@@ -67,7 +139,59 @@ def send_hellos(self):
 
         for node in self.k_nearest:
             ip, port = self.k_nearest[node]
-            self.comm.send(ip, port, "hello")
+            self.send_hello(ip, port)
+            time.sleep(self.hello_delay)
```

Comment:
There shouldn't be a delay here. Delay will be added in Main application loop in Node.run().